examples/tls.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::Result;
+use scylla::transport::session::{IntoTypedRows, Session};
+use scylla::SessionBuilder;
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+
+use openssl::ssl::{SslContextBuilder, SslFiletype, SslMethod, SslVerifyMode};
+
+// How to run scylla instance with TLS:
+//
+// Edit your scylla.yaml file and add paths to certificates
+// ex:
+// client_encryption_options:
+//     enabled: true
+//     certificate: /etc/scylla/db.crt
+//     keyfile: /etc/scylla/db.key
+//
+// If using docker mount your scylla.yaml file and your cert files with option
+// --volume $(pwd)/tls.yaml:/etc/scylla/scylla.yaml
+//
+// If python returns permission error 13 use "Z" flag
+// --volume $(pwd)/tls.yaml:/etc/scylla/scylla.yaml:Z
+//
+// In your Rust program connect to port 9142 if it wasn't changed
+// Create new SslContextBuilder with SslMethod that is used in your connection
+// Use set_certificate_file method with path to your .crt file and its filetype as arguments
+// Set verification mode
+// Build it and add to scylla-rust-driver's SessionBuilder
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Create connection
+    let uri = env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9142".to_string());
+
+    println!("Connecting to {} ...", uri);
+
+    let certdir = fs::canonicalize(PathBuf::from("./test/certs/scylla.crt"))?;
+    let mut context_builder = SslContextBuilder::new(SslMethod::tls())?;
+    context_builder.set_certificate_file(certdir.as_path(), SslFiletype::PEM)?;
+    context_builder.set_verify(SslVerifyMode::NONE);
```

Comment:
Any particular reason for turning off verification? If the certificates are correct, this example works fine without this line - I verified manually with my pair of local cert + key.

Also, I'm experiencing issues with the cert+key generated in your test/certs, Scylla boots, but doesn't open the TLS port due to not being able to decrypt the certificate:
```
WARN  2021-03-24 08:58:45,333 [shard 0] cql_server - Exception loading {/home/sarna/.ccm/scylla-1/node1/conf/scylla.crt, /home/sarna/.ccm/scylla-1/node1/conf/scylla.key}: std::system_error (error GnuTLS:-24, Decryption has failed.)
WARN  2021-03-24 08:58:45,334 [shard 0] cql_server - Exception loading {/home/sarna/.ccm/scylla-1/node1/conf/scylla.crt, /home/sarna/.ccm/scylla-1/node1/conf/scylla.key}: std::system_error (error GnuTLS:-24, Decryption has failed.)
```
@piodul could you verify if it works fine? If it doesn't, here's a pair that works, feel free to replace them with mine:
scylla.key:
```
-----BEGIN RSA PRIVATE KEY-----
MIIJKAIBAAKCAgEApkmL77dy2Y/r4pj2WTklF7YYmmT807ZW4tc7hC2JPGB3KCA6
p7DAcdEwLdzbZLR7qFJgLNjPa60tOok1rfZHRftbhyvIC6rc8+mE6L/FL8tHZNMC
X4imPYKyM9LoPX2nHIshRNAKREfLxHDvA2fDyHryAhJ7WeB+dFrwzJKHm9Bx6FGD
0iAru3fW/eegedr6Eg48VYdlBvyXz74hjEZUGZ+2FO/bB1ctgdi0wp8cTL6F/534
dyc1maBwdSCYiLribjQPXnrSydTycL2/9vigRIGsBZVZc0GrGCAmioQPjn7c5GBo
1ifsJr2CxKPkKHYykp3cF5k36VLLByth9jDqxggl4vQuc8ngFYehg2PiHjpYbEME
+gxGK/DnOmxL54/WMSVkjOnqUIdovKK4taqkvjXKr2JNdzTe9MebM3LEsC14J6iW
up9aXPZ/OfbkjKn46Ac4qnEIUP7QNkb9gN7aCloV6tCu7HpSGTeKQDiWUQr6QAyR
G35/O63ubI98BAysHb1oK0w2Lttfa3sOk8PVv8tSkUlWIiYamvY+nph/06dSsGoF
u81Xb6qPMBQgeVUhP2zDjLlimZ0KyiMCaiiRHIBmrnYWhVo3w+U5uoLI8Rc/xDxj
1spXA/HHY8gFGMlOrMT504jnEHDfLldlzCfpJ2tJm7BWXNzc6kD5tBYzPdkCAwEA
AQKCAgBkEZhrzvsHoEu01pZRhUkkJspRh3p6sZ1B74J6fai7s9WhSRvjNEPrepbC
0skYpmBQIBqOgn74boPsWQv7v1yO6b/f7WH6W32SLzJzmoxAwdUaAL3J3gMkuxUg
iMiH0KkTTUrdDVFQW+BtPAUqCHTSneIrixG+OZH/1YZZpciD5QHXa86nUgLfUzQH
Ii/5hOS+nuCCG/OZK8L0Gpg3wU71CJ7bpKgrl7PkExRclsHCEVm0RU1z7x2bcOic
I9yHU+fVp/Cjy8gz+7lWTWpLZr2T3HE3/vOQn7El2Js4+lYeS4wEx1vvnKDJg3xB
d9ti2hZ/Ew6TCPqTp+ZUTzApprBxOUOxfSnZejLPEXMMKuE/KtI7s971r72iWT0w
8S0M2Wsa7tXCCfheFNNhb7EpL1mDimLhN7+YzDKBQDs/hjtGeTzs3d/xhcfWpWl4
zyyaBRe4uhcddw7pIw9SeVM8V8EU1mQZO5RcCUyy9i4ROiKT7fci1Vm03FFY31T/
gIEHPOpkj/87Yh1vdhgWC2P0cvtqs8reQZQxmZakBuwe+/3RF47gLTd9WLPzMJHK
uqhMGW73AVllfll/RklvwlcioagCt1EcXps2YV7iWN0ss5F8oQQgtzqZO8zYDepT
l8sfVCbUFPqYS8a9edMqfbkddAe+lDjVVwmQDG0FrRMt+N1YXQKCAQEA22t4+lry
x43fvXAhltRPIKWSRDw6SQCp1yHkj++SbF7nR5TMUNGSPueY8hwtz3y9tB6ylSml
1XK2WV3+LdN0CCFpqyNQEiRtAooDWRJMwVxOWCscCcJc1guAuUGb5pRsSdsuuQH9
Xd5Foqv3ebUlTkDTmyD/ETBFq6zcASZ62ZSOiYn2bPVHm18o6LH6Vz41HxfbPi7U
sIcE6/q5nxCqP9U5r4bS5YLHN2wNedYwkE/ptcuqxblLvNmglkZPELF7fHKI6U2f
pW6CROyYwi7v88NApwxdryw81fZVTdtP1yXFoYtuerU8QqvCuUeYc5aa98/z+lyY
ZsuoluBYnoG9EwKCAQEAwgJz/hNbzViOfqFG6mQHdIMbSAtRSYsDfmrXuX2p9DEp
4QbPZJM1l/2dx4II+p0ggGJFeIzH51Yw31/nfZdFZhPt4dh/oToJcQ8Q0kx0g5SV
xwTVz2gRwjGtYJI7CNc/GlsZ+x4blVzaXzaUP4AwFtNyjfS/XWXGll6S/k0Sw7VO
1b/RzBE0FLpsDYd17n3QMmiJmmAVwtggoGblC4aqTQNryLTwAWqbr1x3807eVkgL
6oR6wKD92wvo8ibESJJWeGJxxP/1+bZ3dT4sPlg/vCM+mCTOug+JLDk3BPnUEDSt
FXVHqwmOinx7P7BdsadCWyniFD7gfv375EQjsJXS4wKCAQB1ug037nJSpYmFX9N1
1MX70zpOSC5RmvyFhBbZL4ZVhPn8ixawyjD4dVS3mmzt5ZT3Pn080RK8GodCHc9S
ESExd5P1kcXVZhvi/9/Z+5qZKIiUf9TUW6dp5NQVEeGu7rXHTsWb2Dh0E90y8aef
cpTojpm6oxKXy1mmSeHk8Bt8AFwV6aMBRP09nOobJ9sU0ji4TRmCHKhbaqY2YtkT
QmLGSwZfaixfnjJ7RwtX8pct5uKTewpOeiPr2Kxz1Jd5cnC6Dc3tt4tPvKJCjttL
XNm75Wgb+OOPKGIfvsJ4OpJ08jw+d8w4JGOHthaNSJ/AbO8KxVW3X9UZWm5DPW3u
5ddnAoIBAEaace/ix6PhOGoo8w/Vu0jRdZowVDK5XlohM8STUMhHJAB6Y/sr/3MC
nuq+Ddp2WB+go7uNMCqmGT8RPcNcy6qsUgPAsCHSZyE35F1D6ZsA9cjpn43NR7ku
LvRYAxfijVpgEQJsIImosue3ixfY5KdJ6AR1LzckFLMgU8or7ko+ryPmnMjw8c0h
UrVRuL9EFgJiPMFrej/gO29kQrKBHXTNa+osvrMlQub8dcKbzxoDOiLsd2emrvuK
SAnnxcLLTqCc4t5YcAGiP7LxH1TI9NjY0qy3qGvZv4fSgo1TBbOoF90ZQy2hWrhX
ONsiqdY/Ekzu3xGTGEvezk5x5bMbsxUCggEBAMzqO6NFLF4Gazcy8jKmDKc3Xiql
pTuuw9UCku5OhofSeVNmGKMFDK3v1Axw8etct6hUeRwMvT0QQoC6JQBDU5Km+hLv
32rl18DzmzQqTwTfhOYV8kc0Hyhz1525/peEKrFIZxpw5BaJcR0pd4tB/GxB0+hM
xbt+HgGSjRQ0giomDng4/Y7RCg9wypW88ntwLpY28SlEycdmef2kNb+NpzXZJi4L
CBNazDiv6504bXQiuP9nPj8eIa94Bn37BDFOCrPbU16HbbCGx1LAETGIYQBZH8c9
m2FIMf/b1OBC8Uzh3f/F71Jbn54jESbCbll8mixeUQ+iyBgWPcYlZb8c8Q8=
-----END RSA PRIVATE KEY-----
```
scylla.crt:
```
-----BEGIN CERTIFICATE-----
MIIFtTCCA50CFHXUfp4rO8gXpx7AtTSt9uOXJM+yMA0GCSqGSIb3DQEBCwUAMIGW
MQswCQYDVQQGEwJTRTESMBAGA1UECAwJU3RvY2tob2xtMRIwEAYDVQQHDAlTdG9j
a2hvbG0xEjAQBgNVBAoMCWxvY2FsaG9zdDESMBAGA1UECwwJbG9jYWxob3N0MRIw
EAYDVQQDDAlsb2NhbGhvc3QxIzAhBgkqhkiG9w0BCQEWFHBvc3RtYXN0ZXJAbG9j
YWxob3N0MB4XDTIxMDIyNTE2MTcwN1oXDTIyMDIyNTE2MTcwN1owgZYxCzAJBgNV
BAYTAlNFMRIwEAYDVQQIDAlTdG9ja2hvbG0xEjAQBgNVBAcMCVN0b2NraG9sbTES
MBAGA1UECgwJbG9jYWxob3N0MRIwEAYDVQQLDAlsb2NhbGhvc3QxEjAQBgNVBAMM
CWxvY2FsaG9zdDEjMCEGCSqGSIb3DQEJARYUcG9zdG1hc3RlckBsb2NhbGhvc3Qw
ggIiMA0GCSqGSIb3DQEBAQUAA4ICDwAwggIKAoICAQCmSYvvt3LZj+vimPZZOSUX
thiaZPzTtlbi1zuELYk8YHcoIDqnsMBx0TAt3NtktHuoUmAs2M9rrS06iTWt9kdF
+1uHK8gLqtzz6YTov8Uvy0dk0wJfiKY9grIz0ug9facciyFE0ApER8vEcO8DZ8PI
evICEntZ4H50WvDMkoeb0HHoUYPSICu7d9b956B52voSDjxVh2UG/JfPviGMRlQZ
n7YU79sHVy2B2LTCnxxMvoX/nfh3JzWZoHB1IJiIuuJuNA9eetLJ1PJwvb/2+KBE
gawFlVlzQasYICaKhA+OftzkYGjWJ+wmvYLEo+QodjKSndwXmTfpUssHK2H2MOrG
CCXi9C5zyeAVh6GDY+IeOlhsQwT6DEYr8Oc6bEvnj9YxJWSM6epQh2i8ori1qqS+
NcqvYk13NN70x5szcsSwLXgnqJa6n1pc9n859uSMqfjoBziqcQhQ/tA2Rv2A3toK
WhXq0K7selIZN4pAOJZRCvpADJEbfn87re5sj3wEDKwdvWgrTDYu219rew6Tw9W/
y1KRSVYiJhqa9j6emH/Tp1KwagW7zVdvqo8wFCB5VSE/bMOMuWKZnQrKIwJqKJEc
gGaudhaFWjfD5Tm6gsjxFz/EPGPWylcD8cdjyAUYyU6sxPnTiOcQcN8uV2XMJ+kn
a0mbsFZc3NzqQPm0FjM92QIDAQABMA0GCSqGSIb3DQEBCwUAA4ICAQCfds3MWqtF
vquNDGUEGRylxmTRdMkiNgHioDJDQNJV3v4W+MlAR6KK3weMniXCu33pLQZNOLPK
+7L24rzeYJBlUo4aPIVWwnqneT/Ayo52yCd2NO1SGuuJ9ilgoOSja9eNzuw4jxfy
C4rKvJbCwA7ShOnnLBVgcFd6hww3R73erzo95MMGs24xy/Ioz5CfZl7qq8q4/djw
sF92FLo64sRtuMEXbSGh1EX4SNkT7HoUchonRa2sEscMES7YIAehqFdDfEwoYJJP
N1WIWD6F/JOLTb/oDazDPABQNPbeId/ZLyZFZESBMJOYhMa3qeNHwPI7TzUCvxyq
IchvMmF2Ycy0cBWwM2nwIdf2+Mo5EpshmUN5IOjtR+h0Nj7mK9klnRHbapuf7pw+
Q4/EqScfko3uJK4GYEleVJhAi+Kwru/3ry5pzr6RxkCm/GF0v174E8Ff22OJ4GFW
ZHxv9Ah8O02/Ceh94/XowVhyObJEsBFbooVPi9Jywd/VnPQa0H3QzjfQK9IlU4Dm
zCP+kfM7hIwKLHiOcxoc7cT7v8EPKg/EkxihdXDtQ8OBYN6DzEVP+aOPfvuwrgaw
xMEEKchCSZZyHFN2J8DiI796EQNFlLxUog6zCF8wMWJP+Ds5MoRThiGQgHD3bLdi
eFzZXE81ykF1q9HsbT/nAu0pB54DQxmkZQ==
-----END CERTIFICATE-----
```